File: localWeather/src/main/java/de/interaapps/localweather/ForecastWeather.java
package de.interaapps.localweather;

import com.kwabenaberko.openweathermaplib.models.threehourforecast.ThreeHourForecast;

public class ForecastWeather {

    private ThreeHourForecast threeHourForecast;

    public ForecastWeather(ThreeHourForecast threeHourForecast) {
        this.threeHourForecast = threeHourForecast;
    }

    public String[] getDescriptions() {
        String[] descriptions = new String[]{""};
        for(int i = 0; i < threeHourForecast.getList().size(); i++) {
            for(int j = 0; j < threeHourForecast.getList().get(i).getWeatherArray().size(); j++) {
                descriptions[i] = threeHourForecast.getList().get(i).getWeatherArray().get(j).getDescription();
            }
        }
        return descriptions;
    }

    public String[] getIcons() {
        String[] icons = new String[]{""};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            for(int j = 0; j < threeHourForecast.getList().get(i).getWeatherArray().size(); j++) {
                icons[i] = threeHourForecast.getList().get(i).getWeatherArray().get(j).getIcon();
            }
        }
        return icons;
    }

    public String[] getMains() {
        String[] mains = new String[]{""};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            for(int j = 0; j < threeHourForecast.getList().get(i).getWeatherArray().size(); j++) {
                mains[i] = threeHourForecast.getList().get(i).getWeatherArray().get(j).getMain();
            }
        }
        return mains;
    }

    public long[] getIds() {
        long[] ids = new long[]{0};
        for(int i = 0; i < threeHourForecast.getList().size(); i++) {
            for(int j = 0; j < threeHourForecast.getList().get(i).getWeatherArray().size(); j++) {
                ids[i] = threeHourForecast.getList().get(i).getWeatherArray().get(j).getId();
            }
        }
        return ids;
    }

    public double[] getTemps() {
        double[] temps = new double[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            temps[i] = threeHourForecast.getList().get(i).getMain().getTemp();
        }
        return temps;
    }

    public double[] getMaxTemps() {
        double[] maxTemps = new double[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            maxTemps[i] = threeHourForecast.getList().get(i).getMain().getTempMax();
        }
        return maxTemps;
    }

    public double[] getMinTemps() {
        double[] minTemps = new double[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            minTemps[i] = threeHourForecast.getList().get(i).getMain().getTempMin();
        }
        return minTemps;
    }

    public double[] getTempKfs() {
        double[] tempsKf = new double[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            tempsKf[i] = threeHourForecast.getList().get(i).getMain().getTempKf();
        }
        return tempsKf;
    }

    public double[] getGrndLevels() {
        double[] grnds = new double[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            grnds[i] = threeHourForecast.getList().get(i).getMain().getGrndLevel();
        }
        return grnds;
    }

    public double[] getHumidities() {
        double[] humidities = new double[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            humidities[i] = threeHourForecast.getList().get(i).getMain().getHumidity();
        }
        return humidities;
    }

    public double[] getPressures() {
        double[] pressures = new double[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            pressures[i] = threeHourForecast.getList().get(i).getMain().getPressure();
        }
        return pressures;
    }

    public double[] getSeaLevels() {
        double[] seas = new double[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            seas[i] = threeHourForecast.getList().get(i).getMain().getSeaLevel();
        }
        return seas;
    }

    public double[] getWindSpeeds() {
        double[] wind = new double[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            wind[i] = threeHourForecast.getList().get(i).getWind().getSpeed();
        }
        return wind;
    }

    public double[] getWindDegs() {
        double[] wind = new double[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            wind[i] = threeHourForecast.getList().get(i).getWind().getDeg();
        }
        return wind;
    }

    public double[] getClouds() {
        double[] clouds = new double[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            clouds[i] = threeHourForecast.getList().get(i).getClouds().getAll();
        }
        return clouds;
    }

    public String[] getCountries() {
        String[] countries = new String[]{""};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            countries[i] = threeHourForecast.getList().get(i).getSys().getCountry();
        }
        return countries;
    }

    public double[] getMessages() {
        double[] messages = new double[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            messages[i] = threeHourForecast.getList().get(i).getSys().getMessage();
        }
        return messages;
    }

    public String[] getPods() {
        String[] pods = new String[]{""};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            pods[i] = threeHourForecast.getList().get(i).getSys().getPod();
        }
        return pods;
    }

    public long[] getSunrises() {
        long[] sunrises = new long[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            sunrises[i] = threeHourForecast.getList().get(i).getSys().getSunrise();
        }
        return sunrises;
    }

    public long[] getSunsets() {
        long[] sunsets = new long[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            sunsets[i] = threeHourForecast.getList().get(i).getSys().getSunset();
        }
        return sunsets;
    }

    public long[] getDts() {
        long[] dts = new long[]{0};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            dts[i] = threeHourForecast.getList().get(i).getDt();
        }
        return dts;
    }

    public String[] getDtTexts() {
        String[] dtTexts = new String[]{""};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            dtTexts[i] = threeHourForecast.getList().get(i).getDtTxt();
        }
        return dtTexts;
    }

    public String[] getRains() {
        String[] rains = new String[]{""};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            rains[i] = threeHourForecast.getList().get(i).getRain().toString();
        }
        return rains;
    }

    public String[] getSnows() {
        String[] snows = new String[]{""};
        for (int i = 0; i < threeHourForecast.getList().size(); i++) {
            snows[i] = threeHourForecast.getList().get(i).getSnow().toString();
        }
        return snows;
    }

    public int getCnt() {
        return threeHourForecast.getCnt();
    }

    public String getCod() {
        return threeHourForecast.getCod();
    }

    public double getMessage() {
        return threeHourForecast.getMessage();
    }

    public String getName() {
        return threeHourForecast.getCity().getName();
    }

    public double getLat() {
        return threeHourForecast.getCity().getCoord().getLat();
    }

    public double getLon() {
        return threeHourForecast.getCity().getCoord().getLon();
    }

    public String getCountry() {
        return threeHourForecast.getCity().getCountry();
    }

    private long getId() {
        return threeHourForecast.getCity().getId();
    }

    private long getPopulation() {
        return threeHourForecast.getCity().getPopulation();
    }
}

File: localWeather/src/main/java/de/interaapps/localweather/ForecastWeather.kt
package de.interaapps.localweather

import com.kwabenaberko.openweathermaplib.models.threehourforecast.ThreeHourForecast

class ForecastWeather(private val threeHourForecast: ThreeHourForecast) {

    val descriptions: Array<Array<String>>
        get() {
            val description: MutableList<Array<String>> = mutableListOf()
            for (i in 0 until threeHourForecast.list.size) {
                val desc = mutableListOf<String>()
                for (forecast in threeHourForecast.list[i].weatherArray) {
                    desc.add(forecast.description)
                }
                description.add(desc.toTypedArray())
            }
            return description.toTypedArray()
        }

    val icons: Array<Array<String>>
        get() {
            val icon: MutableList<Array<String>> = mutableListOf()
            for(i in 0 until threeHourForecast.list.size) {
                val ico = mutableListOf<String>()
                for(forecast in threeHourForecast.list[i].weatherArray) {
                    ico.add(forecast.icon)
                }
                icon.add(ico.toTypedArray())
            }
            return icon.toTypedArray()
        }

    val mains: Array<Array<String>>
        get() {
            val main: MutableList<Array<String>> = mutableListOf()
            for(i in 0 until threeHourForecast.list.size) {
                val mai = mutableListOf<String>()
                for(forecast in threeHourForecast.list[i].weatherArray) {
                    mai.add(forecast.main)
                }
                main.add(mai.toTypedArray())
            }
            return main.toTypedArray()
        }

    val ids: Array<Array<Long>>
        get() {
            val id: MutableList<Array<Long>> = mutableListOf()
            for(j in 0 until threeHourForecast.list.size) {
                val i = mutableListOf<Long>()
                for(forecast in threeHourForecast.list[j].weatherArray) {
                    i.add(forecast.id)
                }
                id.add(i.toTypedArray())
            }
            return id.toTypedArray()
        }

    val temperatures: Array<Double>
        get() {
            val temp = mutableListOf<Double>()
            for(forecast in threeHourForecast.list) {
                temp.add(forecast.main.temp)
            }
            return temp.toTypedArray()
        }

    val maxTemperatures: Array<Double>
        get() {
            val temp = mutableListOf<Double>()
            for(forecast in threeHourForecast.list) {
                temp.add(forecast.main.tempMax)
            }
            return temp.toTypedArray()
        }

    val minTemperatures: Array<Double>
        get() {
            val temp = mutableListOf<Double>()
            for(forecast in threeHourForecast.list) {
                temp.add(forecast.main.tempMin)
            }
            return temp.toTypedArray()
        }

    val kfTemperatures: Array<Double>
        get() {
            val temp = mutableListOf<Double>()
            for(forecast in threeHourForecast.list) {
                temp.add(forecast.main.tempKf)
            }
            return temp.toTypedArray()
        }

    val groundLevels: Array<Double>
        get() {
            val ground = mutableListOf<Double>()
            for(forecast in threeHourForecast.list) {
                ground.add(forecast.main.grndLevel)
            }
            return ground.toTypedArray()
        }

    val humidities: Array<Double>
        get() {
            val humidity = mutableListOf<Double>()
            for(forecast in threeHourForecast.list) {
                humidity.add(forecast.main.humidity)
            }
            return humidity.toTypedArray()
        }

    val pressures: Array<Double>
        get() {
            val pressure = mutableListOf<Double>()
            for(forecast in threeHourForecast.list) {
                pressure.add(forecast.main.pressure)
            }
            return pressure.toTypedArray()
        }

    val seaLevels: Array<Double>
        get() {
            val sea = mutableListOf<Double>()
            for(forecast in threeHourForecast.list) {
                sea.add(forecast.main.seaLevel)
            }
            return sea.toTypedArray()
        }

    val windSpeeds: Array<Double>
        get() {
            val wind = mutableListOf<Double>()
            for(forecast in threeHourForecast.list) {
                wind.add(forecast.wind.speed)
            }
            return wind.toTypedArray()
        }

    val windAngles: Array<Double>
        get() {
            val wind = mutableListOf<Double>()
            for(forecast in threeHourForecast.list) {
                wind.add(forecast.wind.deg)
            }
            return wind.toTypedArray()
        }

    val clouds: Array<Double>
        get() {
            val cloud = mutableListOf<Double>()
            for(forecast in threeHourForecast.list) {
                cloud.add(forecast.clouds.all)
            }
            return cloud.toTypedArray()
        }

    val countries: Array<String>
        get() {
            val country = mutableListOf<String>()
            for(forecast in threeHourForecast.list) {
                country.add(forecast.sys.country)
            }
            return country.toTypedArray()
        }

    val messages: Array<Double>
        get() {
            val message = mutableListOf<Double>()
            for(forecast in threeHourForecast.list) {
                message.add(forecast.sys.message)
            }
            return message.toTypedArray()
        }

    val pods: Array<String>
        get() {
            val pod = mutableListOf<String>()
            for(forecast in threeHourForecast.list) {
                pod.add(forecast.sys.pod)
            }
            return pod.toTypedArray()
        }

    val sunrises: Array<Long>
        get() {
            val sun = mutableListOf<Long>()
            for(forecast in threeHourForecast.list) {
                sun.add(forecast.sys.sunrise)
            }
            return sun.toTypedArray()
        }

    val sunsets: Array<Long>
        get() {
            val sun = mutableListOf<Long>()
            for(forecast in threeHourForecast.list) {
                sun.add(forecast.sys.sunset)
            }
            return sun.toTypedArray()
        }

    val dts: Array<Long>
        get() {
            val dt = mutableListOf<Long>()
            for(forecast in threeHourForecast.list) {
                dt.add(forecast.dt)
            }
            return dt.toTypedArray()
        }

    val dtTexts: Array<String>
        get() {
            val dt = mutableListOf<String>()
            for(forecast in threeHourForecast.list) {
                dt.add(forecast.dtTxt)
            }
            return dt.toTypedArray()
        }

    val rains: Array<String>
        get() {
            val rain = mutableListOf<String>()
            for(forecast in threeHourForecast.list) {
                rain.add(forecast.rain.toString())
            }
            return rain.toTypedArray()
        }

    val snows: Array<String>
        get() {
            val snow = mutableListOf<String>()
            for(forecast in threeHourForecast.list) {
                snow.add(forecast.snow.toString())
            }
            return snow.toTypedArray()
        }

    val cnt: Int = threeHourForecast.cnt

    val cod: String = threeHourForecast.cod

    val message: Double = threeHourForecast.message

    val cityName: String = threeHourForecast.city.name

    val cityLatitude: Double = threeHourForecast.city.coord.lat

    val cityLongitude: Double = threeHourForecast.city.coord.lon

    val country: String = threeHourForecast.city.country

    val cityPopulation: Long = threeHourForecast.city.population
}